consensus/reactor.go

```diff
@@ -696,31 +696,30 @@ func (conR *ConsensusReactor) gossipVotesForHeight(logger log.Logger, rs *cstype
 			return true
 		}
 	}
-
-	if prs.Step <= cstypes.RoundStepCommit && prs.Round != -1 && prs.Round <= rs.Round {
-		// If there are POLPrevotes to send...
-		if prs.ProposalPOLRound != -1 {
-			if polPrevotes := rs.Votes.Prevotes(prs.ProposalPOLRound); polPrevotes != nil {
-				if ps.PickSendVote(polPrevotes) {
-					logger.Debug("Picked rs.Prevotes(prs.ProposalPOLRound) to send",
-						"round", prs.ProposalPOLRound)
-					return true
-				}
-			}
-		}
-
-		// If there are prevotes to send...
+	// If there are prevotes to send...
+	if prs.Step <= cstypes.RoundStepPrevoteWait && prs.Round != -1 && prs.Round <= rs.Round {
```

Comment:
@jaekwon You made the following comment: 
> As an example, when a peer is on the Propose step, we'll still be sending all the POL prevotes until the peer 
> jumps to the Prevote step, in which case we'll send Prevote votes... until we run out of prevotes, and then we'll > go back to sending the remaining POL prevotes. 

I fully agree with you here. While a node is in the Propose step POL prevotes have high priority as they can be seen as part of Proposal message in the broader sense. Imagine the scenario where a proposer p has locked value v in round r, and this is the highest locked value among correct processes. Imagine that process q has locked value `v' != v` in round `r' < r`. In this case, it is critical for termination that q receives POL prevotes so he can unlock v' and vote for v. So we want to prioritise on sending POL prevotes while process is in Propose step. Once process moves to Prevote step, POL prevotes have less priority so I agree we should give more priority to Prevotes and then later Precommits. We should still send POL prevotes if there is nothing else with higher priority to be sent. Note that in this version we are never prioritising POL prevotes. So we probably need to add one more rule that will check if `prs.Step <= cstypes.RoundStepPropose` and send POL prevotes then as the first rule.